addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1878,7 +1883,16 @@ const SnippetOptionWidget = Widget.extend({
                 if (cssValues.length === 2 && weUtils.areCssValuesEqual(cssValues[1], cssValues[0], params.cssProperty, this.$target)) {
                     cssValues.pop();
                 }
-                return cssValues.join(' ');
+                let value = cssValues.join(' ');
+                if (params.cssProperty === 'box-shadow') {
+                    // matching HEXA | HEX | HEX-shorthand color notations.
+                    const hexColor = value.match(/#([A-Fa-f0-9]{8}|[A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})/g);
+                    const color = hexColor ? hexColor[0] : value.slice(value.indexOf('rgb'), value.indexOf(')') + 1);
+                    const inset = value.indexOf('inset') >= 0 ? 'inset' : '';
```

Comment:
I know I am annoying :smile:  but technically it could be in `hsl` format too. I think it would be more generic to:
1. split on space (by removing the spaces of rgb first like you did somewhere else)
2. search for 'inset' match to know if you have to put inset or not in the final result
3. search the color as the value which does not begin by a number
4. form the final value

Don't you agree?

Reply:
Sounds good, although on the browsers I checked (chrome, edge, firefox) it returned rgba, it's definitely better.